src/databricks/labs/ucx/assessment/dashboards.py

```diff
@@ -0,0 +1,370 @@
+        schema: str,
+        *,
+        include_dashboard_ids: list[str] | None = None,
+        include_query_ids: list[str] | None = None,
+    ):
+        super().__init__(sql_backend, "hive_metastore", schema, "lakeview_dashboards", Dashboard)
+        self._ws = ws
+        self._include_dashboard_ids = include_dashboard_ids or []
+        self._include_query_ids = include_query_ids or []
+
+    def _crawl(self) -> Iterable[Dashboard]:
+        dashboards = []
+        for sdk_dashboard in self._list_dashboards():
+            if sdk_dashboard.dashboard_id is None:
+                continue
+            dashboard = Dashboard.from_sdk_lakeview_dashboard(sdk_dashboard)
+            dashboards.append(dashboard)
+        return dashboards
+
+    def _list_dashboards(self) -> list[SdkLakeviewDashboard]:
+        if self._include_dashboard_ids:
+            return self._get_dashboards(*self._include_dashboard_ids)
+        try:
+            # If the API listing limit becomes an issue in testing, please see the `:class:RedashDashboardCrawler`
+            # for an example on how to implement a (debug) rate limit
+            return list(self._ws.lakeview.list())  # TODO: Add dashboard summary view?
+        except DatabricksError as e:
+            logger.warning("Cannot list Lakeview dashboards", exc_info=e)
+            return []
+
+    def _get_dashboards(self, *dashboard_ids: str) -> list[SdkLakeviewDashboard]:
+        dashboards = []
+        for dashboard_id in dashboard_ids:
+            dashboard = self._get_dashboard(dashboard_id)
+            if dashboard:
+                dashboards.append(dashboard)
+        return dashboards
+
+    def _get_dashboard(self, dashboard_id: str) -> SdkLakeviewDashboard | None:
+        try:
+            return self._ws.lakeview.get(dashboard_id)
+        except DatabricksError as e:
+            logger.warning(f"Cannot get Lakeview dashboard: {dashboard_id}", exc_info=e)
+            return None
+
+    def _try_fetch(self) -> Iterable[Dashboard]:
+        for row in self._fetch(f"SELECT * FROM {escape_sql_identifier(self.full_name)}"):
+            yield Dashboard(*row)
+
+    def list_queries(self, dashboard: Dashboard | None = None) -> Iterator[Query]:
+        """List queries.
+
+        Args:
+            dashboard (DashboardType | None) : List queries for dashboard. If None, list all queries.
+                Defaults to None.
+
+        Note:
+            This public method does not adhere to the common crawler layout, still, it is implemented to avoid/postpone
+            another crawler for the queries by retrieving the queries every time they are requested.
+
+            Different to the Redash crawler, Lakeview queries are part of the (serialized) dashboard definition.
+        """
+        sdk_dashboards = []
+        if dashboard:
+            sdk_dashboard = self._get_dashboard(dashboard_id=dashboard.id)
+            if sdk_dashboard:
+                sdk_dashboards.append(sdk_dashboard)
```

Comment:
What do you think of this?
```suggestion
        if dashboard:
            sdk_dashboard = self._get_dashboard(dashboard_id=dashboard.id)
            sdk_dashboards = [sdk_dashboard] if sdk_dashboard else []
```